app/src/main/java/me/guillaumin/android/osmtracker/db/DataHelper.java

```diff
@@ -65,6 +68,9 @@
 	 */
 	private Context context;
 
+	/** Statistics for all existing tracks */
+	private static TreeMap<Long, TrackStatistics> tracksStatistics = new TreeMap<Long, TrackStatistics> ();
```

Comment:
Probably not if that's this fast.

One thing to consider to avoid storing state in the DataHelper class are ContentObservers. They allow you to register a function to be called when something changes in the DB, so perhaps doing that in `TrackListAdapter` would allow you to update the values as they change?
* http://developer.android.com/reference/android/database/ContentObserver.html
* http://www.grokkingandroid.com/use-contentobserver-to-listen-to-changes/